ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -593,26 +593,31 @@ private JoinNode buildJoin(final Join root, final String prefix, final boolean i
       );
     }
 
-    final boolean isForeignKeyJoin =
+    final Optional<ColumnReferenceExp> fkColumnName =
         verifyJoin(root.getInfo(), preRepartitionLeft, preRepartitionRight);
 
     final boolean finalJoin = prefix.isEmpty();
 
-    final JoinKey joinKey = buildJoinKey(root);
+    final JoinKey joinKey;
+    if (fkColumnName.isPresent()) {
+      joinKey = buildForeignJoinKey(root, fkColumnName.get());
```

Comment:
Why do we need to pass `root` and `fkColumnName`? The output key should be the key of the left input, so it seems to be sufficient to pass `root.getLeft()` into this method?